llama_index/embeddings/cohereai.py

```diff
@@ -0,0 +1,113 @@
+from typing import Any, List, Optional
+
+from llama_index.bridge.pydantic import Field
+from llama_index.callbacks import CallbackManager
+from llama_index.embeddings.base import DEFAULT_EMBED_BATCH_SIZE, BaseEmbedding
+
+# Constants for default values
+DEFAULT_MODELS_REQUIRING_INPUT_TYPE = [
+    "embed-english-v3.0",
+    "embed-english-light-3.0",
+    "embed-multilingual-v3.0",
+    "embed-multilingual-light-v3.0",
+]
+
+DEFAULT_INPUT_TYPES = [
+    "search_query",
+    "search_document",
+    "classification",
+    "clustering",
+]
+DEFAULT_TRUNCATE_OPTIONS = ["START", "END", "NONE"]
```

Comment:
I wonder if we can follow a similar pattern to how we handle this type of defining "modes" and "model" as we did in `OpenAIEmbeddings`. That is, using Enum and Dicts.